frontend/src/shared/helpers.tsx

```diff
@@ -1038,72 +1077,99 @@ export function prepareSampleMetadataForAgGrid(samples: Sample[]) {
 
 export function prepareSampleCohortDataForAgGrid(samples: Sample[]) {
   return samples.map((s) => {
-    const cohorts = s.cohortsHasCohortSampleConnection?.edges;
-    const cohortDates = cohorts?.flatMap((c) => {
-      return c.node.hasCohortCompleteCohortCompletes.map((cc) => {
-        return cc.date;
-      });
+    const sampleMetadata = s.hasMetadataSampleMetadata[0];
+    const tempoData = extractTempoFromSample(s);
+
+    return {
+      primaryId: sampleMetadata.primaryId,
+      cmoSampleName: sampleMetadata.cmoSampleName,
+      hasStatusStatuses: sampleMetadata.hasStatusStatuses,
+      revisable: s.revisable,
+      ...tempoData,
+    };
+  });
+}
+
+export function prepareCombinedSampleDataForAgGrid(samples: Sample[]) {
+  return samples.map((s) => {
+    const sampleMetadata = s.hasMetadataSampleMetadata[0];
+    const tempoData = extractTempoFromSample(s);
+
+    return {
+      revisable: s.revisable,
+      ...sampleMetadata,
+      ...tempoData,
+    };
+  });
+}
+
+function extractTempoFromSample(s: Sample) {
```

Comment:
The code inside this new function is mostly the same code found in the previous definition of `prepareSampleCohortDataForAgGrid()`. I generalized it into this named function so that it could be used for both (1) the existing Cohort Sample view and (2) the new Samples page.